Source/Generators/LMCCavitySignalGenerator.cc

```diff
@@ -531,50 +553,63 @@ namespace locust
-        std::vector<std::deque<double>>().swap(tLocalFIRfrequencyBuffer);  // release memory
-        std::vector<std::deque<double>>().swap(tLocalElementFIRBuffer);
+    					// fix-me:  We may need more precise nFilterBinsRequired.
+    					std::vector<std::deque<double>> tLocalFIRfrequencyBuffer = fInterface->FIRfrequencyBufferCopy;  // copy from Kass buffer.
+    					std::vector<std::deque<double>> tLocalElementFIRBuffer = fInterface->ElementFIRBufferCopy;
+    					double tFirSample = GetCavityFIRSample(tKassParticleXP, tLocalFIRfrequencyBuffer, tLocalElementFIRBuffer, fInterface->nFilterBinsRequired, fInterface->dtFilter);
+    					std::vector<std::deque<double>>().swap(tLocalFIRfrequencyBuffer);  // release memory
+    					std::vector<std::deque<double>>().swap(tLocalElementFIRBuffer);
 
-        for(int channelIndex = 0; channelIndex < nChannels; ++channelIndex)  // one channel per probe.
-        {
-        	sampleIndex = channelIndex*signalSize*aSignal->DecimationFactor() + index;  // which channel and which sample
-
-        	// TO-DO:  Implement modeScalingFactor to calculate signal amplitude at probe/channel using
-        	// H-field and probe inductance.  Probe inductance is presently set to a fake value of 1.0
-        	// in LMCPowerCombiner.cc
-        	// double modeScalingFactor = GetModeScalingFactor(tKassParticleXP, channelIndex);
-        	double modeScalingFactor = 1.0; // override for now.
-        	// end TO-DO
-
-        	// This scaling factor includes a 50 ohm impedance that is expected in signal processing, as well
-        	// as other factors as defined above.
-        	double totalScalingFactor = sqrt(50.) * unitConversion * impedanceFactor * modeScalingFactor * dotProductFactor * modeAmplitude;
-        	fPowerCombiner->AddOneModeToCavityProbe(aSignal, tFirSample, fphiLO, totalScalingFactor, fPowerCombiner->GetCavityProbeInductance(), sampleIndex);
-        }
+    					for(int channelIndex = 0; channelIndex < nChannels; ++channelIndex)  // one channel per probe.
+    					{
+    						sampleIndex = channelIndex*signalSize*aSignal->DecimationFactor() + index;  // which channel and which sample
+
+    						// TO-DO:  Implement modeScalingFactor to calculate signal amplitude at probe/channel using
+    						// H-field and probe inductance.  Probe inductance is presently set to a fake value of 1.0
+    						// in LMCPowerCombiner.cc
+    						// double modeScalingFactor = GetModeScalingFactor(tKassParticleXP, channelIndex);
+    						double modeScalingFactor = 1.0; // override for now.
+    						// end TO-DO
+
+    						// This scaling factor includes a 50 ohm impedance that is expected in signal processing, as well
+    						// as other factors as defined above.
+    						double totalScalingFactor = sqrt(50.) * unitConversion * impedanceFactor * modeScalingFactor * dotProductFactor * modeAmplitude;

```

Comment:
Again I know many of these factors are placeholders for future work, but it isn't intuitive to me that there are separate factors for the 50 ohm impedance as well as an `impedanceFactor`. Would it make sense to rename the latter to something like `expansionCoefficient` if it's based off of the equations from Foundations for ME? I think the current variable name comes from the older version of the code where you were using the Jackson definition of an actual mode impedance term.

Reply:
The 50 ohms is the receiver impedance assumed and applied in katydid post processing.  So I think it is distinct from the field coefficients here.  In the meantime I agree we should rename impedanceFactor to be expansionCoefficient.  This change has been committed.